### PR TITLE
Auth login/logout 

### LINKED ISSUE #94: login for teachers and students- user authentication
finish user authentication for teachers (bring up correct classes); reroute to survey for students

### PR DESCRIPTION
**Pull request by kelvinjue**
fixed #94

### CODE REVIEW

File: .vscode/launch.json [hunk 1]
```diff
@@ -0,0 +1,14 @@
+{
```
Comment (bchee) [inline .vscode/launch.json]: Don't commit this file

File: client/src/components/Dashboard/AddClassBox/AddClassBox.jsx [hunk 2]
```diff
@@ -4,6 +4,7 @@ import Modal from 'react-modal';
 import PropTypes from 'prop-types';
 import Item from './Item';
 import plus from './plusImage.png';
+// import { IncomingMessage } from 'http';
```
Comment (bchee) [inline client/src/components/Dashboard/AddClassBox/AddClassBox.jsx]: Delete

File: client/src/components/Logout/Logout.jsx [hunk 3]
```diff
@@ -3,9 +3,9 @@ import './Logout.css';
 
 function Logout() {
   return (
-    <button type="submit" className="logout">
+    <a href="http://localhost:8080/auth/logout/" className="logout">
```
Comment (bchee) [inline client/src/components/Logout/Logout.jsx]: You don't need the local host part

File: client/src/components/Login/Login.jsx [hunk 4]
```diff
@@ -11,7 +11,7 @@ class Login extends Component {
           <h1 className="welcome">Welcome!</h1>
           <div className="welcome-sub">Get started here.</div>
           <div className="button-wrapper">
-            <a href="/" className="signin">
+            <a href="http://localhost:8080/auth/google/" className="signin">
```
Comment (bchee) [inline client/src/components/Login/Login.jsx]: Remove local host

File: server/routes/auth.js [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+  req.logout();
+  req.session = null;
+  res.redirect('http://localhost:3000/login');
+});
+
+router.get(
+  '/google',
+  passport.authenticate('google', {
+    scope: [
+      'https://www.googleapis.com/auth/userinfo.profile',
```
Comment (bchee) [inline server/routes/auth.js]: Shorten these to just 'profile' and 'email'

File: server/server.js [hunk 6]
```diff
@@ -0,0 +1,66 @@
+const express = require('express');
+
```
Comment (bchee) [inline server/server.js]: Don't commit this file. Make sure before you commit to do a git diff or git status and make sure to only commit files for master and not test files

File: server/routes/index.js [hunk 7]
```diff
@@ -4,6 +4,22 @@ const db = require('../db/index');
 
 const router = new Router();
 
+router.get('/getUsers', async (req, res) => {
```
Comment (bchee) [inline server/routes/index.js]: Okay sorry I gave you guys this idea to make a `/getUsers` route. Truth is that this route is unnecessary and there is a better way to do this. Luckily, the change is very easy and should make your code much cleaner. Delete this route and just make it a function that returns the userid of users logged in. Then in routes like `/classes` you don't need to take in the userId from the client you can just call this within the `GET /classes` route for express.
```
async function getUsers(req, res) {
  try {
    if (Object.keys(req.session).length === 0) {
      res.redirect('/login');
      return false;
    } else {
      const query = await db.query(
        'SELECT id FROM users WHERE users.token = $1',
        [req.session.passport.user.token]
      );
      return query.rows[0].id.toString();
    }
  } catch (error) {
    console.log(error.stack);
    return false;
  }
}
```

File: client/src/components/Login/Login.jsx (hunk 4 above)
Comment (kelvinjue) [reply client/src/components/Login/Login.jsx]: i think we need it to run the app

File: client/src/components/Logout/Logout.jsx (hunk 3 above)
Comment (kelvinjue) [reply client/src/components/Logout/Logout.jsx]: same as above, might need localhost

File: client/src/components/Dashboard/Dashboard.jsx [hunk 8]
```diff
@@ -33,11 +33,24 @@ class Dashboard extends Component {
   }
 
   async componentWillMount() {
-    const classes = await fetch('/api/classes');
-    const classesJSON = await classes.json();
-    this.setState({
-      classList: classesJSON
-    });
+    const { history } = this.props;
+    const classRoute = await fetch('/api/getUsers', { redirect: 'follow' });
```
Comment (bchee) [inline client/src/components/Dashboard/Dashboard.jsx]: getUsers?

**Pull request merged**